core/dataset.cpp

```diff
@@ -252,6 +266,14 @@ void Dataset::setAttr(const std::string &attrName, Variable attr) {
   m_attrs.insert_or_assign(attrName, std::move(attr));
 }
 
+/// Set (insert or replace) the masks for the given label name.
+///
+/// Note that the label name has no relation to names of data items.
+void Dataset::setMasks(const std::string &labelName, Variable masks) {
+  setDims(masks.dims());
+  m_masks.insert_or_assign(labelName, std::move(masks));
+}
```

Comment:
Do we consider the mask as a variable, which only can have `boolean` values? If so, the check of `dtype` is needed for the argument.

Reply:
Yeah, it should only be a Variable of bools. I will add the check!